fastplotlib/subplot.py

```diff
@@ -59,21 +60,34 @@ def __init__(
 
         self._animate_funcs = list()
 
-        self.renderer.add_event_handler(self._produce_rect, "resize")
+        self.renderer.add_event_handler(self.set_viewport_rect, "resize")
 
-    def _produce_rect(self, *args):#, w, h):
-        i, j = self.position
+        self.docked_viewports = dict()
+        for pos in ["left", "top", "right", "bottom"]:
+            self.docked_viewports[pos] = _DockedViewport(self, pos, size=0)
+
+        self.set_viewport_rect()
 
+    def get_rect(self):
+        i, j = self.position
         w, h = self.renderer.logical_size
 
         spacing = 2  # spacing in pixels
 
-        self.viewport.rect = [
+        return np.array([
```

Comment:
I really need to write this better, like:

x_pos =...
y_pos =...
width =...

viewport.rect = (x, y, w, h)